sips/sip-021/sip-021-nakamoto.md

```diff
@@ -0,0 +1,1051 @@
+`TenureChange` transactions.
+
+4. **Download and validate the continuity of each block sequence between
+consecutive block commits**.  Now that the node knows the number of blocks
+between two consecutive winning block-commits, as well as the hashes of the
+first and last block in this sequence, the node can do this in a bound amount of
+space and time.  There is no risk of a malicious node serving an endless stream
+of well-formed but invalid blocks to a booting-up node, because the booting-up
+node knows exactly how many blocks to expect and knows what hashes they must
+have.
+
+5. **Concurrently processes newly-downloaded blocks in reward cycle _R+1_** to
+build up its tenure of the blockchain.
+
+6. **Repeat once the PoX anchor block for R+2 has been downloaded and
+processed**
+
+This bootup procedure is amenable to activating this proposal from Stacks 2.x,
+as long as it happens on a reward cycle boundary.
+
+When the node has synchronized to the latest reward cycle, it would run this
+algorithm to discover new tenures within the current reward cycle until it
+reaches the chain tip.  Once it has processed all blocks as of the last
+sortition, it continues to keep pace with the current miner by receiving new
+blocks broadcasted through the peer network by Stackers once they accept the
+next Stacks block.
+
+## Block Structure
+
+Nakamoto Stacks blocks will have a different wire format than Stacks blocks and
+microblocks today.  In particular, the header will contain:
+
+* A version number (1 byte)
+* The length of the chain as of this tip (8 bytes, big-endian)
+* The total BTC spent producing this tip (8 bytes, big-endian)
+* The `StacksBlockId` of the parent Stacks block (32 bytes)
+* The Bitcoin block hash of the Bitcoin block that triggered the start of the
+  tenure in which they were mined (32 bytes)
+* A recoverable ECDSA signature from the tenure's miner (65 bytes)
+* The root of a SHA512/256 Merkle tree constructed over all of its contained
+  transactions (32 bytes)
+* The SHA512/256 root hash of the MARF once all of the contained transactions
+  are processed (32 bytes)
+* A Schnorr signature collectively generated by the set of Stackers over this
+  block (65 bytes)
+
+As before, the Stacks blockchain will continue to calculate a consensus hash for
+each sortition.  The `StacksBlockId` of a Stacks block would simply be the
+SHA512/256 of the block's associated sortition's consensus hash and the hash of
+the block header.
+
+Absent from this header is the VRF proof, because it only needs to be included
+once per tenure.  Instead, this information will be put into the Nakamoto
+`Coinbase` transaction, which has a different wire format than the current
+`Coinbase` transaction.
+
+All existing Stacks transactions, with the exception of `PoisonMicroblock`
+transactions and the current `Coinbase` transaction, will continue to be
+supported.  However, the anchor mode byte will be ignored.
+
+### Tenure Changes
```

Comment:
We reference tenure change a few times prior to this section. I feel this should either be included in the glossary or defined (at least briefly) prior to its first use in the doc.

Reply:
Added a definition in `Introduction/Design`